server/src/test/java/io/druid/metadata/TestDerbyConnector.java

```diff
@@ -18,31 +18,112 @@
 package io.druid.metadata;
 
 import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
 import io.druid.metadata.storage.derby.DerbyConnector;
 import org.junit.Assert;
+import org.junit.rules.ExternalResource;
 import org.skife.jdbi.v2.DBI;
 import org.skife.jdbi.v2.exceptions.UnableToObtainConnectionException;
 
 import java.sql.SQLException;
+import java.util.UUID;
 
 public class TestDerbyConnector extends DerbyConnector
 {
+  public static class DerbyConnectorRule extends ExternalResource
+  {
+    private TestDerbyConnector connector;
+    private final Supplier<MetadataStorageTablesConfig> dbTables;
+    private final MetadataStorageConnectorConfig connectorConfig;
+
+    public DerbyConnectorRule()
+    {
+      this(Suppliers.ofInstance(MetadataStorageTablesConfig.fromBase("druidTest")));
+    }
+
+    public DerbyConnectorRule(
+        Supplier<MetadataStorageTablesConfig> dbTables
+    )
+    {
+      this.dbTables = dbTables;
+      this.connectorConfig = new MetadataStorageConnectorConfig()
+      {
+        @Override
+        public String getConnectURI()
+        {
+          return connector.getJdbcUri();
+        }
+      };
+    }
+
+    @Override
+    protected void before() throws Throwable
+    {
+      connector = new TestDerbyConnector(Suppliers.ofInstance(connectorConfig), dbTables);
+      connector.getDBI().open().close(); // create db
+    }
+
+    @Override
+    protected void after()
+    {
+      connector.tearDown();
+    }
+
+    public TestDerbyConnector getConnector()
+    {
+      return connector;
+    }
+
+    public MetadataStorageConnectorConfig getMetadataConnectorConfig()
+    {
+      return connectorConfig;
+    }
+
+    public Supplier<MetadataStorageTablesConfig> metadataTablesConfigSupplier()
+    {
+      return dbTables;
+    }
+  }
+
+  private final String jdbcUri;
+
   public TestDerbyConnector(
       Supplier<MetadataStorageConnectorConfig> config,
       Supplier<MetadataStorageTablesConfig> dbTables
   )
   {
-    super(config, dbTables, new DBI("jdbc:derby:memory:druidTest;create=true"));
+    this(config, dbTables, "jdbc:derby:memory:druidTest" + dbSafeUUID());
```

Comment:
I'm assuming that , appending the uuid is really making the difference .. different test runs will have their own db name now?